第5,6章終了

shmn7iii opened the pull request:
Heroku：https://morning-tor-81081.herokuapp.com/

shmn7iii committed:
delete example_user.rb

ほんとは4章末で消すはずだけどレビューもらうために残してた。ので削除。

shmn7iii committed:
5.1.1 navigation

shmn7iii committed:
5.1.2 Bootstrap and custom CSS

shmn7iii committed:
5.1.3 partial

shmn7iii committed:
exercise 5.2.1

shmn7iii committed:
5.3.1 contact page

shmn7iii committed:
5.3.2 root URL

shmn7iii committed:
5.3.3 named root

shmn7iii committed:
delete unnecessary space

shmn7iii committed:
5.3.4 test link

shmn7iii committed:
exercise 5.3.4

shmn7iii committed:
5.4.1 users controller

shmn7iii committed:
exercise 5.4.1

shmn7iii committed:
5.4.2 signup URL

shmn7iii committed:
exercise 5.4.2

shmn7iii committed:
update readme and separate documents

学びを章ごとに分けてdoc配下に設置しました

shmn7iii committed:
delete .keep

オプションつけたと思ったのになんか入ってた。なんで。

shmn7iii committed:
update readme

shmn7iii committed:
6.1.1 generate User model

shmn7iii committed:
6.2.1 valid test

shmn7iii committed:
6.2.2 presence test

shmn7iii committed:
6.2.3 length test

shmn7iii committed:
6.2.4 format test

shmn7iii committed:
exercise 6.2.4

shmn7iii committed:
6.2.5 uniqueness test

shmn7iii committed:
exercise 6.2.5

shmn7iii committed:
6.3.1 migrate

shmn7iii committed:
6.3.2 has_secure_password

shmn7iii committed:
6.3.3 password minimum length

shmn7iii committed:
update readme

shmn7iii committed:
Merge branch 'develop' into feature/chap5

shmn7iii committed:
update home.html.erb

<a> link -> link_to

shmn7iii committed:
Merge pull request #6 from shmn7iii/feature/chap5

develop <- feature/chap5

shmn7iii committed:
Merge pull request #7 from shmn7iii/feature/chap6

develop <- feature/chap6

shmn7iii committed:
update readme

p0pemaru commented on app/assets/stylesheets/users.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
これ消しておいて問題無いです！

p0pemaru commented on app/helpers/users_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
+end
```
これも削除して大丈夫ですよ〜

p0pemaru commented on db/migrate/20220307053344_create_users.rb:
```diff
@@ -0,0 +1,10 @@
+  def change
+    create_table :users do |t|
+      t.string :name
+      t.string :email
```
```suggestion
      t.string :email, null: false
```

FYI
バリデーションをつけるのが一般的です！（今回は tutorial に書いてある通り進めていただいても大丈夫です）

p0pemaru commented on app/models/user.rb:
```diff
@@ -0,0 +1,10 @@
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: { case_sensitive: false }
+  has_secure_password
+  validates :password, presence: true, length: { minimum: 6 }
```
```suggestion
  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
  has_secure_password
  
  validates :name,  presence: true, length: { maximum: 50 }
  validates :email, presence: true, length: { maximum: 255 },
                    format: { with: VALID_EMAIL_REGEX },
                    uniqueness: { case_sensitive: false }
  validates :password, presence: true, length: { minimum: 6 }
```

`validates:`行が点在してみにくいので、まとめたほうが良さそうですね！

p0pemaru commented on app/views/layouts/_footer.html.erb:
```diff
@@ -0,0 +1,13 @@
+<footer class="footer">
+  <small>
+    The <a href="https://railstutorial.jp/">Ruby on Rails Tutorial</a>
+    by <a href="https://www.michaelhartl.com/">Michael Hartl</a>
```
ここも`link_to`使ってもらって大丈夫ですよ！

p0pemaru commented on app/views/static_pages/home.html.erb:
```diff
@@ -1,6 +1,17 @@
+  <%= link_to "Sign up now!", signup_path, class: "btn btn-lg btn-primary" %>
+</div>
+
+<%#= link_to image_tag("kitten.jpeg", alt: "kitten", width: "200px"),
+                      "https://rubyonrails.org/" %>
```
これは不要かな？

shmn7iii replied:
前の章の演習問題で扱った内容をコメントアウトしてます！

p0pemaru commented on db/schema.rb:
```diff
@@ -10,6 +10,15 @@
 
-ActiveRecord::Schema.define(version: 0) do
+ActiveRecord::Schema.define(version: 2022_03_07_074552) do
+
```
nits.
ここの空行は取って良いですよ

p0pemaru commented on db/schema.rb:
```diff
@@ -10,6 +10,15 @@
+    t.string "password_digest"
+    t.index ["email"], name: "index_users_on_email", unique: true
+  end
 
```
この空行も同じく

p0pemaru commented on doc/chap6.md:
````diff
@@ -0,0 +1,86 @@
+
+- 「Railsコンソール サンドボックスモード」  
+    ```bash
+    $ rails console --sandbox
````
```suggestion
    $ rails c -s
```
FYI
短縮したエイリアスです

p0pemaru commented on doc/chap6.md:
```diff
@@ -0,0 +1,86 @@
+    `.destroy` は create の逆 ただしメモリ上には残る 戻り値はオブジェクト自身  
+
+- 「User.find / .find_by / .first / .all」  
+    `.find` idを引数にとってオブジェクトを返す  
+    `.find_by` 任意の属性を引数にとってオブジェクトを返す  
+    `.first` DBの最初のオブジェクトを返す  
+    `.all` DBの全てのオブジェクトを返す  
```
FYI
`find`, `find_by`, `find_by!`, `where`の違い（対象のオブジェクトが見つからなかった場合の挙動なども）も良かったら調べてみてください:smile:

p0pemaru commented on test/controllers/users_controller_test.rb:
```diff
@@ -0,0 +1,9 @@
+    get signup_path
+    assert_response :success
+  end
+
```
不要な空行ですね！
rubocop など linter を入れ、保存時に勝手にフォーマットされるようエディタを設定しておくのがおすすめです

p0pemaru commented on test/integration/site_layout_test.rb:
```diff
@@ -0,0 +1,17 @@
+require 'test_helper'
+
+class SiteLayoutTest < ActionDispatch::IntegrationTest
+
```
この空行も不要です！

p0pemaru commented on test/models/user_test.rb:
```diff
@@ -0,0 +1,77 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+
```
この空行も不要です！

shmn7iii committed:
delete app/assets/stylesheets/users.scss

shmn7iii committed:
re-format app/model/user.rb

shmn7iii committed:
replace <a> with link_to

shmn7iii committed:
delete unnecessary line

shmn7iii committed:
add rubocop

以下の記事を参考
https://qiita.com/d0ne1s/items/a3566aca023d11f2bfc4
https://qiita.com/noraworld/items/3e5179d2ec3313b25c20

shmn7iii committed:
update readme

レビュー反映

p0pemaru commented on doc/chap6.md:
```diff
@@ -0,0 +1,136 @@
+    https://qiita.com/noraworld/items/3e5179d2ec3313b25c20  
+
+    あとVS Codeの空白消す設定、markdownの末尾消されるから切ってたけど解決策あったので有効化した  
+    https://qiita.com/otera@github/items/d715f760aab2f6e88e67
```
ちなみに改行スペースは見にくい（改行しているかが見かけ上わからない）ので、`<br>`タグを使ったり、空行を挿入するのが個人的にはおすすめです。
もちろんこの辺の考え方は人それぞれなので、使いやすいと感じる物で問題なしです！

p0pemaru reviewed (approved):
LGTM!

1つだけコメント追記していますが FYI なので読んでもらうだけで大丈夫です！
